### PR TITLE
Require GnuTLS 3.1.7, and make more pcmk_scheduler_t members internal

### PR DESCRIPTION
**Pull request by kgaillot**

### CODE REVIEW

**Comment by kgaillot** [comment]
@nrwahl2 , here's the next set of drops

**Review by nrwahl2 — COMMENTED** [review]
Nice

File: rpm/pacemaker.spec.in [hunk 1]
```diff
@@ -181,7 +181,6 @@
 %global pkgname_bzip2_devel libbz2-devel
 %global pkgname_docbook_xsl docbook-xsl-stylesheets
 %global pkgname_gettext gettext-tools
-%global pkgname_gnutls_devel libgnutls-devel
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply rpm/pacemaker.spec.in:184]: If we use `PKG_CHECK_MODULES` for a dependency in configure.ac, we should use `pkgconfig` for it in the spec (both use pkg-config to locate the dependency)

File: lib/common/remote.c [hunk 2]
```diff
@@ -158,58 +158,6 @@ pcmk__tls_client_handshake(pcmk__remote_t *remote, int timeout_sec,
...
- * \brief Set minimum prime size required by TLS client
- *
- * \param[in] session  TLS session to affect
- */
-static void
-set_minimum_dh_bits(const gnutls_session_t *session)
-{
-    int dh_min_bits;
-
-    pcmk__scan_min_int(pcmk__env_option(PCMK__ENV_DH_MIN_BITS), &dh_min_bits,
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply lib/common/remote.c:172]: oops, meant to

[the remark this answers is not in the document]
Comment (kgaillot) [reply lib/common/remote.c:172]: updated commit

**Review by nrwahl2 — APPROVED** [review]

**Pull request merged**